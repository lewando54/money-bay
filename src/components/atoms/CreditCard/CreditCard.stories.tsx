/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import CreditCard from './CreditCard'

const meta = {
	component: CreditCard,
	argTypes: {
		cardIssuer: { control: 'select', options: ['visa', 'mastercard'] },
        cardName: {control: 'text'},
        cardNumber: {control: 'text'},
        cardExpiryDate: { control: 'date' },
        cardBalance: { control: 'number' },
        cardCurrency: { control: 'object' }
	}
} satisfies Meta<typeof CreditCard>

export default meta
type Story = StoryObj<typeof meta>

export const Visa: Story = {
	args: {
		cardIssuer: 'visa',
        cardName: 'bay platinum',
        cardNumber: '1234 **** **** 1234',
        cardExpiryDate: new Date('2023-12-19'),
        cardBalance: 4863.27,
        cardCurrency: {iso: 'usd', symbol: '$', left: true}
	}
}

export const Mastercard: Story = {
	args: {
		cardIssuer: 'mastercard',
        cardName: 'bay platinum',
        cardNumber: '1234 **** **** 1234',
        cardExpiryDate: new Date('2023-12-19'),
        cardBalance: 4863.27,
        cardCurrency: {iso: 'usd', symbol: '$', left: true}
	}
}

export const CurrencyLeft: Story = {
	args: {
		cardIssuer: 'mastercard',
        cardName: 'bay platinum',
        cardNumber: '1234 **** **** 1234',
        cardExpiryDate: new Date('2023-12-19'),
        cardBalance: 4863.27,
        cardCurrency: {iso: 'usd', symbol: '$', left: true}
	}
}

export const CurrencyRight: Story = {
	args: {
		cardIssuer: 'mastercard',
        cardName: 'bay platinum',
        cardNumber: '1234 **** **** 1234',
        cardExpiryDate: new Date('2023-12-19'),
        cardBalance: 4863.27,
        cardCurrency: {iso: 'pln', symbol: 'z≈Ç', left: false}
	}
}