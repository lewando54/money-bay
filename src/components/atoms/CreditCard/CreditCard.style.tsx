import { Dimensions, StyleSheet } from 'react-native'

const {width, height} = Dimensions.get('screen')

export default StyleSheet.create({
    cardView: {
        flexDirection: 'row', 
        aspectRatio: 16/9, 
        width: width * 0.7733,
        borderRadius: 10
    },
    blurView: {
        flex: 1,
        width: '100%',
        height: '100%',
        borderRadius: 10,
    },
    cardBorderMask: {
        borderWidth: 1,
        borderColor: 'black',
        backgroundColor: 'transparent',
        flex: 1,
        borderRadius: 10,
        justifyContent: 'center',
        alignItems: 'center'
    },
    linearGradient: { 
        flex: 1,
        height: '100%',
        opacity: 0.5
    },
    cardIssuerLogo: {
        width: 40,
        aspectRatio: 16/9,
        resizeMode: 'contain'
    },
    cardName: {
        letterSpacing: 3,
        fontSize: 14,
        color: 'white'
    },
    separator: {
        backgroundColor: 'white',
        width: 0.5,
        height: 21,
        opacity: 0.5
    },
    cardNumber: {
        fontSize: 12
    },
    secondaryText: {
        fontSize: 8,
        opacity: 0.5,
        textTransform: 'uppercase'
    },
    infoBoxesWrapper: {
        flexDirection: 'row', 
        marginTop: 'auto',
        justifyContent: 'space-between'
    },
    infoBox: {
        gap: 5
    },
    infoBoxData: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        gap: 7
    },
    balanceBox: {
        flexDirection: 'row'
    },
    balanceInt: {
        fontSize: 24
    },
    cardExpiry: {
        opacity: 0.8
    }
})