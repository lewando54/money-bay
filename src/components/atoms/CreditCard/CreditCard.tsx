import React from 'react'
import { Image,Text, View } from 'react-native'
import MaskedView from '@react-native-masked-view/masked-view'
import { BlurView } from 'expo-blur'
import { LinearGradient } from 'expo-linear-gradient'

import { getIssuerLogo,TCardIssuer } from '../../../utils/cardIssuers'
import { TCurrency as TCurrency } from '../../../utils/currency'

import globalThemeStyle from '../../../styling/GlobalTheme.style'
import creditCardStyle from './CreditCard.style'

interface ICreditCardProps {
    cardIssuer?: TCardIssuer
    cardName?: string
    cardNumber?: string
    cardExpiryDate?: Date
    cardBalance?: number
    cardCurrency?: TCurrency
    testID?: string
}

export default function CreditCard({
    cardIssuer = 'visa',
    cardName = 'bay platinum',
    cardNumber = '1234 **** **** 1234',
    cardExpiryDate = new Date('2023-12-19'),
    cardBalance = 4863.27,
    cardCurrency = {iso: 'usd', symbol: '$', left: true},
    testID = 'creditcard'
}: ICreditCardProps){
    // if MaskedView creates an error, go to node_modules/@react-native-masked-view/masked-view/types/index.d.ts
    // And change:
    /*
        declare class MaskedViewComponent extends React.Component<MaskedViewProps> {}
        declare const MaskedViewBase: ReactNative.Constructor<
        ReactNative.NativeMethods
        > &
        typeof MaskedViewComponent;
        export default class MaskedView extends MaskedViewBase {}
    */
    // to:
    /*
        declare class MaskedViewComponent extends React.Component<MaskedViewProps> {}
        export default class MaskedView extends MaskedViewComponent {}
    */
    return (
        <View style={creditCardStyle.cardView} testID={testID}>
            <MaskedView
                style={{width: '100%'}}
                maskElement={
                    <View style={creditCardStyle.cardBorderMask}>
                        <BlurView intensity={100} style={creditCardStyle.blurView}>
                        </BlurView>
                    </View>
                }
            >
                <LinearGradient 
                    style={creditCardStyle.linearGradient}
                    colors={['#161966', '#AEB1FF', '#FF97B4']}
                    start={{x: 0, y: 0}}
                    end={{x: 1, y: 1}}
                >
                </LinearGradient>
            </MaskedView>
            <View style={{position: 'absolute', paddingVertical: 20, flex: 1, height: '100%', width: '100%', paddingHorizontal: 20, gap: 14}}>
                <View style={{flexDirection: 'row', alignItems: 'center', gap: 13}}>
                    <Image 
                        style={creditCardStyle.cardIssuerLogo} 
                        source={getIssuerLogo(cardIssuer)}>
                    </Image>
                    <View style={creditCardStyle.separator}></View>
                    <Text style={[globalThemeStyle.text_Regular, creditCardStyle.cardName]}>{cardName}</Text>
                </View>
                <View>
                    <Text style={[globalThemeStyle.text_Regular, creditCardStyle.cardNumber]}>
                        {cardNumber}
                    </Text>
                </View>
                <View style={creditCardStyle.infoBoxesWrapper}>
                    <View style={creditCardStyle.infoBox}>
                        <Text style={[globalThemeStyle.text_Regular, creditCardStyle.secondaryText]}>
                            Balance
                        </Text>
                        <View style={creditCardStyle.infoBoxData}>
                            <Text style={[globalThemeStyle.text_Bold, {fontSize: 23}, !cardCurrency.left && {display: 'none'}]}>
                                {cardCurrency.symbol}
                            </Text>
                            <View style={creditCardStyle.balanceBox}>
                                <Text style={[globalThemeStyle.text_Bold, {marginBottom: 0}, creditCardStyle.balanceInt]}>
                                    {Math.floor(cardBalance).toLocaleString('en-US', {minimumIntegerDigits: 2}).replace(',', ' ')}.
                                </Text>
                                <Text style={[globalThemeStyle.text_Bold, {height: '100%', paddingTop: 11}]}>
                                    {((cardBalance * 100) % 100).toLocaleString('en-US', {minimumIntegerDigits: 2})}
                                </Text>
                                <Text style={[globalThemeStyle.text_Bold, {fontSize: 12, marginLeft: 7, paddingTop: 12.7}, cardCurrency.left && {display: 'none'}]}>
                                    {cardCurrency.symbol}
                                </Text>
                            </View>
                        </View>
                    </View>
                    <View style={creditCardStyle.infoBox}>
                        <Text style={[globalThemeStyle.text_Regular, creditCardStyle.secondaryText]}>
                            Expire
                        </Text>
                        <View style={creditCardStyle.infoBoxData}>
                            <View style={creditCardStyle.balanceBox}>
                                <Text style={[globalThemeStyle.text_Medium, creditCardStyle.cardExpiry]}>
                                    {cardExpiryDate.getMonth() + 1}/{('' + cardExpiryDate.getFullYear()).substring(2)}
                                </Text>
                            </View>
                        </View>
                    </View>
                </View>
            </View>
        </View>
    )
}