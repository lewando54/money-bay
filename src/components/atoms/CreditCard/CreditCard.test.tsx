import React from 'react'
import renderer from 'react-test-renderer'
import CreditCard from './CreditCard'
describe('CreditCard', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<CreditCard />)
        expect(tree).toMatchSnapshot()
    })

    it('should render mastercard logo when chosen', () => {
        const tree = renderer.create(<CreditCard cardIssuer='mastercard'/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render currency symbol on the right when chosen', () => {
        const tree = renderer.create(<CreditCard cardCurrency={{iso: 'pln', symbol: 'zÅ‚', left: false}}/>)
        expect(tree).toMatchSnapshot()
    })
})
