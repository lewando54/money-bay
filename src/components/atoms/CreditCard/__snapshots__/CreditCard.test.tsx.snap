// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard should render correctly 1`] = `
<View
  style={
    {
      "aspectRatio": 1.7777777777777777,
      "borderRadius": 10,
      "flexDirection": "row",
      "width": "80%",
    }
  }
>
  <RNCMaskedView
    style={
      {
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
              },
              {
                "borderRadius": 10,
                "flex": 1,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <ViewManagerAdapter_ExpoBlurView
            proxiedProperties={
              {
                "intensity": 100,
                "tint": "default",
              }
            }
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4279638374,
            4289638911,
            4294940596,
          ],
          "endPoint": [
            1,
            1,
          ],
          "locations": undefined,
          "startPoint": [
            0,
            0,
          ],
        }
      }
      style={
        {
          "flex": 1,
          "height": "100%",
          "opacity": 0.5,
        }
      }
    />
  </RNCMaskedView>
  <View
    style={
      {
        "flex": 1,
        "gap": 14,
        "height": "100%",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 13,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "aspectRatio": 1.7777777777777777,
            "resizeMode": "contain",
            "width": 40,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "white",
            "height": 21,
            "opacity": 0.5,
            "width": 0.5,
          }
        }
      />
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 3,
            },
          ]
        }
      >
        bay platinum
      </Text>
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "fontSize": 12,
            },
          ]
        }
      >
        1234 **** **** 1234
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": "auto",
        }
      }
    >
      <View
        style={
          {
            "gap": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "fontSize": 8,
                "opacity": 0.5,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Balance
        </Text>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "gap": 7,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "MulishBold",
                },
                {
                  "fontSize": 23,
                },
                false,
              ]
            }
          >
            $
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "marginBottom": 0,
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
            >
              4863
              .
            </Text>
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "height": "100%",
                    "paddingTop": 11,
                  },
                ]
              }
            >
              27
            </Text>
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "fontSize": 12,
                    "marginLeft": 7,
                    "paddingTop": 12.7,
                  },
                  {
                    "display": "none",
                  },
                ]
              }
            >
              $
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "gap": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "fontSize": 8,
                "opacity": 0.5,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Expire
        </Text>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "gap": 7,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishMedium",
                  },
                  {
                    "opacity": 0.8,
                  },
                ]
              }
            >
              12
              /
              23
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`CreditCard should render currency symbol on the right when chosen 1`] = `
<View
  style={
    {
      "aspectRatio": 1.7777777777777777,
      "borderRadius": 10,
      "flexDirection": "row",
      "width": "80%",
    }
  }
>
  <RNCMaskedView
    style={
      {
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
              },
              {
                "borderRadius": 10,
                "flex": 1,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <ViewManagerAdapter_ExpoBlurView
            proxiedProperties={
              {
                "intensity": 100,
                "tint": "default",
              }
            }
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4279638374,
            4289638911,
            4294940596,
          ],
          "endPoint": [
            1,
            1,
          ],
          "locations": undefined,
          "startPoint": [
            0,
            0,
          ],
        }
      }
      style={
        {
          "flex": 1,
          "height": "100%",
          "opacity": 0.5,
        }
      }
    />
  </RNCMaskedView>
  <View
    style={
      {
        "flex": 1,
        "gap": 14,
        "height": "100%",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 13,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "aspectRatio": 1.7777777777777777,
            "resizeMode": "contain",
            "width": 40,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "white",
            "height": 21,
            "opacity": 0.5,
            "width": 0.5,
          }
        }
      />
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 3,
            },
          ]
        }
      >
        bay platinum
      </Text>
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "fontSize": 12,
            },
          ]
        }
      >
        1234 **** **** 1234
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": "auto",
        }
      }
    >
      <View
        style={
          {
            "gap": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "fontSize": 8,
                "opacity": 0.5,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Balance
        </Text>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "gap": 7,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "MulishBold",
                },
                {
                  "fontSize": 23,
                },
                {
                  "display": "none",
                },
              ]
            }
          >
            zł
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "marginBottom": 0,
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
            >
              4863
              .
            </Text>
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "height": "100%",
                    "paddingTop": 11,
                  },
                ]
              }
            >
              27
            </Text>
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "fontSize": 12,
                    "marginLeft": 7,
                    "paddingTop": 12.7,
                  },
                  false,
                ]
              }
            >
              zł
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "gap": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "fontSize": 8,
                "opacity": 0.5,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Expire
        </Text>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "gap": 7,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishMedium",
                  },
                  {
                    "opacity": 0.8,
                  },
                ]
              }
            >
              12
              /
              23
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`CreditCard should render mastercard logo when chosen 1`] = `
<View
  style={
    {
      "aspectRatio": 1.7777777777777777,
      "borderRadius": 10,
      "flexDirection": "row",
      "width": "80%",
    }
  }
>
  <RNCMaskedView
    style={
      {
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "black",
            "borderRadius": 10,
            "borderWidth": 1,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
              },
              {
                "borderRadius": 10,
                "flex": 1,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <ViewManagerAdapter_ExpoBlurView
            proxiedProperties={
              {
                "intensity": 100,
                "tint": "default",
              }
            }
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4279638374,
            4289638911,
            4294940596,
          ],
          "endPoint": [
            1,
            1,
          ],
          "locations": undefined,
          "startPoint": [
            0,
            0,
          ],
        }
      }
      style={
        {
          "flex": 1,
          "height": "100%",
          "opacity": 0.5,
        }
      }
    />
  </RNCMaskedView>
  <View
    style={
      {
        "flex": 1,
        "gap": 14,
        "height": "100%",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 13,
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "aspectRatio": 1.7777777777777777,
            "resizeMode": "contain",
            "width": 40,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "white",
            "height": 21,
            "opacity": 0.5,
            "width": 0.5,
          }
        }
      />
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 3,
            },
          ]
        }
      >
        bay platinum
      </Text>
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "fontSize": 12,
            },
          ]
        }
      >
        1234 **** **** 1234
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": "auto",
        }
      }
    >
      <View
        style={
          {
            "gap": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "fontSize": 8,
                "opacity": 0.5,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Balance
        </Text>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "gap": 7,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontFamily": "MulishBold",
                },
                {
                  "fontSize": 23,
                },
                false,
              ]
            }
          >
            $
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "marginBottom": 0,
                  },
                  {
                    "fontSize": 24,
                  },
                ]
              }
            >
              4863
              .
            </Text>
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "height": "100%",
                    "paddingTop": 11,
                  },
                ]
              }
            >
              27
            </Text>
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishBold",
                  },
                  {
                    "fontSize": 12,
                    "marginLeft": 7,
                    "paddingTop": 12.7,
                  },
                  {
                    "display": "none",
                  },
                ]
              }
            >
              $
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "gap": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "MulishRegular",
              },
              {
                "fontSize": 8,
                "opacity": 0.5,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Expire
        </Text>
        <View
          style={
            {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "gap": 7,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "white",
                    "fontFamily": "MulishMedium",
                  },
                  {
                    "opacity": 0.8,
                  },
                ]
              }
            >
              12
              /
              23
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
